homeassistant/bootstrap.py

```diff
@@ -539,9 +543,16 @@ async def _async_set_up_integrations(
                 STAGE_1_TIMEOUT, cool_down=COOLDOWN_TIME
             ):
                 await async_setup_multi_components(hass, stage_1_domains, config)
+
         except asyncio.TimeoutError:
             _LOGGER.warning("Setup timed out for stage 1 - moving forward")
 
+    # If there is a database upgrade in progress the recorder
+    # queue can exaust the available memory if we allow stage 2
+    # to start. We wait until the upgrade is completed before
+    # starting.
+    await async_wait_for_recorder_full_startup(hass)
+
```

Comment:
This will go away in the next PR https://github.com/home-assistant/core/pull/49036